src/test/java/solution/EnumBoardTest.java

```diff
@@ -0,0 +1,282 @@
+        unsolvedArray[4] = new char[]{'4', '.', '.', '8', '.', '3', '.', '.', '1'};
+        unsolvedArray[5] = new char[]{'7', '.', '.', '.', '2', '.', '.', '.', '6'};
+        unsolvedArray[6] = new char[]{'.', '6', '.', '.', '.', '.', '2', '8', '.'};
+        unsolvedArray[7] = new char[]{'.', '.', '.', '4', '1', '9', '.', '.', '5'};
+        unsolvedArray[8] = new char[]{'.', '.', '.', '.', '8', '.', '.', '7', '9'};
+
+        unsolvedBoard = new EnumBoard(unsolvedArray);
+    }
+
+    @Test
+    void Should_ThrowIllegalArgumentException_When_ArrayDimensions0x0() {
+        char[][] empty = new char[0][0];
+        Exception thrown =
+                assertThrows(IllegalArgumentException.class,
+                             () -> {
+                                 new EnumBoard(empty);
+                             });
+        assertTrue(thrown.getMessage().contains("Illegal board dimensions. Sudoku board needs to be of 9x9 dimensions."));
+    }
+
+    @Test
+    void Should_ThrowIllegalArgumentException_When_ArrayDimensions0x9() {
+        char[][] empty = new char[0][9];
+        Exception thrown =
+                assertThrows(IllegalArgumentException.class,
+                             () -> {
+                                 new EnumBoard(empty);
+                             });
+        assertTrue(thrown.getMessage().contains("Illegal board dimensions. Sudoku board needs to be of 9x9 dimensions."));
+    }
+
+    @Test
+    void Should_ThrowIllegalArgumentException_When_ArrayDimensions9x0() {
+        char[][] empty = new char[9][0];
+        Exception thrown =
+                assertThrows(IllegalArgumentException.class,
+                             () -> {
+                                 new EnumBoard(empty);
+                             });
+        assertTrue(thrown.getMessage().contains("Illegal board dimensions. Sudoku board needs to be of 9x9 dimensions."));
+    }
+
+    @Test
+    void Should_ThrowIllegalArgumentException_When_ArrayDimensions9x10() {
+        char[][] empty = new char[9][10];
+        Exception thrown =
+                assertThrows(IllegalArgumentException.class,
+                             () -> {
+                                 new EnumBoard(empty);
+                             });
+        assertTrue(thrown.getMessage().contains("Illegal board dimensions. Sudoku board needs to be of 9x9 dimensions."));
+    }
+
+    @Test
+    void Should_ThrowIllegalArgumentException_When_ArrayDimensions10x9() {
+        char[][] empty = new char[10][9];
+        Exception thrown =
+                assertThrows(IllegalArgumentException.class,
+                             () -> {
+                                 new EnumBoard(empty);
+                             });
+        assertTrue(thrown.getMessage().contains("Illegal board dimensions. Sudoku board needs to be of 9x9 dimensions."));
+    }
+
+    @Test
+    void Should_ThrowIllegalArgumentException_When_ArrayDimensions10x10() {
```

Comment:
After looking at more of these negative tests on board dimensions, I think this case calls for a parametrized test.